lib/github-pages-health-check/domain.rb

```diff
@@ -165,30 +165,44 @@ def valid_domain?
       def apex_domain?
         return @apex_domain if defined?(@apex_domain)
 
-        return unless valid_domain?
+        return false unless valid_domain?
+
+        return true if dns_zone_soa? && dns_zone_ns?
 
         # PublicSuffix.domain pulls out the apex-level domain name.
         # E.g. PublicSuffix.domain("techblog.netflix.com") # => "netflix.com"
         # It's aware of multi-step top-level domain names:
         # E.g. PublicSuffix.domain("blog.digital.gov.uk") # => "digital.gov.uk"
         # For apex-level domain names, DNS providers do not support CNAME records.
+        #
+        # TODO: Should we even use this here vs allowing DNS to be source of truth?
```

Comment:
I would keep it as a fallback mechanism.

Kind of inline with https://stackoverflow.com/a/16395268.